editoast/src/main.rs

```diff
@@ -118,6 +124,56 @@ fn init_sentry(args: &RunserverArgs) -> Option<ClientInitGuard> {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+struct Claims {
+    sub: String,
+    exp: usize,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Jwk {
+    kty: String,
+    #[serde(rename = "use")]
+    jwt_use: String,
+    n: String,
+    e: String,
+    kid: String,
+    alg: Option<String>,
+}
+
+#[derive(Debug, Deserialize)]
+struct JwksResponse {
+    keys: Vec<Jwk>,
+}
+
+async fn get_jwks(domain: &str) -> Result<Vec<Jwk>, reqwest::Error> {
+    let jwks_uri = format!("{}.well-known/jwks.json", domain);
+    let jwks: JwksResponse = reqwest::get(&jwks_uri).await?.json().await?;
+    Ok(jwks.keys)
+}
+
+async fn validator(
+    req: ServiceRequest,
+    credentials: BearerAuth,
+) -> Result<ServiceRequest, (ActixError, ServiceRequest)> {
+    // let secret = env::var("OSRD_AUTH_ZERO_SECRET").expect("auth0 secret must be set");
+    let domain = env::var("OSRD_AUTH_ZERO_DOMAIN").expect("auth0 domain must be set");
+    let mut validation = Validation::new(Algorithm::RS256);
+    validation.set_issuer(&[&domain]);
+
+    let jwks = get_jwks(&domain).await.expect("Failed to fetch JWKS");
```

Comment:
We should cache the `JwkSet` at startup not in the validator.
To retrieve the `JkwSet` we can use an app_data. Checkout this example https://docs.rs/actix-web-httpauth/0.8.0/actix_web_httpauth/middleware/struct.HttpAuthentication.html#examples-1